### ISSUE TITLE
Add Smart Work Zones Feed

### ISSUE THREAD

**Issue by j-d-b** [usdot-jpo-ode/wzdx#191]
## Background
Currently, the WZDx specification defines a schema and object model for representing work zones in a single GeoJSON file. A valid WZDx GeoJSON file is an instance of a WZDx feed. The WZDx feed contains real time information about work zones, both current and in the near future. The WZDx feed is intended to be consumed by the traveling public, most often through a third-party such as connected vehicle or wayfinding application. 

In 2021, the WZDx Smart Work Zones Subgroup was charted with the goal of standardizing the format of work zone field device information. It became clear through discussion that the consumers desiring the standard for field device information are not the same as those desiring the road event information. For example, the traveling public likely does not care that there is a speed detector on the side of the road—they need to know when and where the work is occurring and what the impact to vehicles is (which is road event information already in the WZDx feed). Thus the field device information should not be added to the WZDx feed.

## Proposal
The consumer of the WZDx feed (represented by the [WZDxFeed](https://github.com/usdot-jpo-ode/wzdx/blob/main/spec-content/objects/WZDxFeed.md) object) is the traveling public and the consumer of the field device information is agencies such as DOTs who are interested in using the field device data for a range of use cases such as combining it with other data sources to generate a WZDx feed. Thus the SWZ subgroup proposed defining a new Smart Work Zone "Field Devices Feed" with its own schema and root object (not within the WZDxFeed) for representing and disseminating field device information.

### Feeds
Feed Name | Producer | Consumer | Uses
--- | --- | --- | ---
WZDx Feed (existing) | Agencies responsible for managing roadways and road work, typically state and local DOTs | Traveling public via mapping companies and CAVs | Route planning; increased awareness; "put work zones on the map"
SWZ Field Devices Feed (new) | SWZ equipment manufacturers | Agencies responsible for managing roadways and permitting work, typically state and local DOTs who are also interested in producing a WZDx Feed | Aids in dynamically generating a WZDx feed; simplifies design process for agencies wanting to interface with equipment manufacturers; reduces effort for manufacturers to conform to different agencies requirements

### Data flow
```
                                                                                 |
                                                                                 | Various
                                                                                 | Additional Inputs
                                                            SWZ                  v
  Deployed Field    Lower-Level   +------------------+  Field Devices  +------------------+            +----------------+
     Devices       Communication  |                  |      Feed       |                  |    WZDx    |                |
+---+ +---+ +---+    Protocols    |    Equipment     | --------------> |     DOT or       |    Feed    |    Traveling   |
|   | |   | |   |  -------------> |   Manufacturer   |                 |  Similar Agency  | ---------> |     Public     |
+---+ +---+ +---+                 |    or Vendor     | --------------> |                  |            |                |
                                  |                  |      WZDx       |                  |            |                |
                                  +------------------+      Feed       +------------------+            +----------------+
```
The new feed would define a schema for a GeoJSON FeatureCollection containing a list of field devices (which would be GeoJSON features), alike to how the WZDx feed is a GeoJSON FeatureCollection that has a list of road events.

## Notes and other thoughts
- The addition of a new feed does not affect the WZDx feed at all—there is no change to the WZDx feed schema
- We may want to consider renaming the WZDx feed to clarify its use and content, if all feeds are considered part of the WZDx effort (perhaps `RoadEventFeed`)—again this would not affect the content or structure of a WZDx feed.
- If "WZDx" and "the WZDx specification" is strictly the public-consumption road event information, this new feed is not "WZDx" and it may be best to house it in a separate repository.
- This is exactly how #181 could be implemented, with another new feed, the "StructureFeed"
- We may want to share the "FeedInfo" (i.e. [RoadEventFeedInfo](https://github.com/usdot-jpo-ode/wzdx/blob/main/spec-content/objects/RoadEventFeedInfo.md)) between feeds, which may require other changes
- The consumer for the WZDx feed in the above diagram is both the DOT and the public, which may be mixing use cases (however, we know Ver-Mac is already using WZDx for this use case)
 
## Next steps
- Contributors, is the proposed field devices feed of value to your agency or company?
- The SWZ subgroup team will create GitHub issues for the field devices models proposed for inclusion in the first version (1.0) of the SWZ Field Devices Feed
- The SWZ subgroup team with work with the working members to determine and finalize the desired properties for each field device 
- The SWZ subgroup team will work will work with the specification update subgroup to refactor the repository organization and READMEs to include the new feed in a clear way and decide how to include the new feed without confusion/clutter

**Comment by seyoung1991** [comment]
The WZDX is exposing several areas that require digital exchange standardization - such as this SWZ as well as the 'Posted Structures' that was brought up in another committee.  In additional to these two extensions (or separate initiatives) spawned by the WZDX initiative, several aspirational comments from state DOTs indicate a pent up demand for more activities such as WZDZ, but in other areas.

**Comment by j-d-b** [comment]
@seyoung1991 exactly, there is demand, but how do we organize the efforts, what do we brand as "WZDx", and do we address them all as "part of WZDx", or should they be more separate (e.g. separate repositories)?

I think they could all be part of a "Transportation Data Exchange" (name for discussion) which has shared/common objects, defines a set of root-level GeoJSON FeatureCollections which each represent a "feed". Each feed would have its own schema.

*Example refactoring of the specification organization to accommodate multiple "feeds" (defined GeoJSON documents) with different use cases and shared models (objects and enumerated types):*

```
transportation-data-exchange/
├─ common/
│  ├─ objects/
│  │  ├─ FeedInfo
│  ├─ enumerated-types/
│  │  ├─ Direction
├─ field-devices/
│  ├─ objects/
│  │  ├─ FieldDevice
│  │  ├─ FieldDevicesFeed
│  ├─ enumerated-types/
│  │  ├─ FieldDeviceType
│  ├─ FieldDeviceFeed.schema.json
├─ structures/
│  ├─ objects/
│  │  ├─ Structure
│  │  ├─ StructuresFeed
│  ├─ enumerated-types/
│  │  ├─ StructureType
│  ├─ StructuresFeed.schema.json
├─ wzdx/
│  ├─ objects/
│  │  ├─ WZDxFeed
│  │  ├─ RoadEvent
│  ├─ enumerated-types/
│  │  ├─ EventType
│  ├─ WZDxFeed.schema.json
```

However, this may be straying too far from the work zones focus of WZDx, especially structures, which have no relation to work zones at all. Field devices are deployed as part of work zones and the data from them can be used to populate a WZDx feed for public consumption, so they fit into the goal of the work zone data initiative.

**Comment by seyoung1991** [comment]
Jason,
I am less concerned with maintaining the overall "Transportation Data Exchange" organizational structure as I am getting these data exchanges efficiently created, propagated, and evolved.  I believe the WZDx needs to be finalized and propagated as expediently as possible, and keeping as close to the original vision and mission as possible (avoid extensions to the core WZDx as is currently discussed - keeping them as separate initiatives).   The group that creates 'the structures' is likely more aligned with freight - so forcing it into the WZDz structure may not serve its benefits well.  Similar thought with the devices interests.  Finding the right group or organization to incubate the initial data exchange likely has the most impact on overall success.   Ultimately adoption (and success of the standard) is governed by procurement, that is the procurement specs call out the data exchange --- so finding the states that are on the leading edge of system procurement will likely do more to drive adoption of new data standards than anything else.

Food for thought,

Stan

**Comment by j-d-b** [comment]
@seyoung1991 WZDx is "finalized" and the feed is in use by several producers and being implemented by many more after the demonstration grants were awarded earlier this year—and I agree about keeping close to the original vision for WZDx, for the content of the public WZDx feed, at least.

For the field devices feed specifically, MassDOT received support of several other agencies and was awarded a WZDx grant to implement what this issue is addressing with models and structure defined by the community. So, the time, interested members, and financial backing to make it happen is ready now.

Thus back to my concern and comments about how this new feed is presented and where it lives/evolves in relation to the public WZDxFeed of work zone road event information, as I feel like that is critical. Is it part of WZDx and should the WZDx repository be refactored to support it (note again no change to the WZDxFeed spec) as I gave an example of above, or should it be in its own repository as a separate initiative? We will discuss this in the next Smart Work Zones Subgroup meeting with the members and I will update this issue with the results of that discussion. I am not sure how which direction I prefer at this point and need more input from current WZDx users and members of all subgroups to propose an implementation.

**Comment by seyoung1991** [comment]
Thanks Jason,
That helps in getting up to speed on the status of the top level WZDx.
Stan

**Comment by j-d-b** [comment]
I agree with @DeraldDudley about needing to discuss with chairs and co-chairs, then members.

Until then, i am thinking about this more and noting my thoughts for reference:

There are two approaches to adding "feeds" and use cases to the WZDx effort, *if we decide they are part of WZDx/this repository*.

## 1. Separating feeds by content
The first is described in my initial comment, which is splitting the feed by the data content, that is, WZDxFeed contains work zone road events (thus my thought about renaming it to RoadEventsFeed), FieldDevicesFeed contains field devices, StructuresFeed contains structures, etc.

In this case, more specifically we would be defining the structure and content of several separate GeoJSON FeatureCollections each representing a collection of a specific feature type (road event, field device, structure, etc.).

The downside of this approach is that use case is not clearly defined. Maybe that is fine and more flexible, or maybe its obfuscating the single purpose of WZDx that is just to implement one use case (work zone info for the public) of the WZDI (which for what its worth, I cannot find any information about online, and hit several broken links). 

## 2. Separating feeds by use case and producer/consumer
The second option is separating the feeds by the use case and producer/consumer in each scenario. With this approach both the WZDxFeed (producer: DOTs/similar; consumer: the public) and the FieldDevicesFeed (producer: vendor/manufacturer; consumer: DOTs/similar) could contain overlapping information (e.g. they could both contain road events—this could help the proposal in #178 stay separate from work zones as well, with another new feed, as *a restriction-notice* is not a work zone). "FieldDevicesFeed" would need to be renamed in this case, to "VendorWorkZoneDataFeed", "SWZInfoFeed", or something similar indicating it is a feed for work zone information from the equipment supplier or implementer. 

This approach would allow leaving the WZDxFeed untouched both in name and use case and the data flow would change from my first comment to the following:

```
                                                                                 |
                                                                                 | Various
                                                                                 | Additional Inputs
                                                                                 v
  Deployed Field    Lower-Level   +------------------+      SWZ        +------------------+            +----------------+
     Devices       Communication  |                  |      Info       |                  |    WZDx    |                |
+---+ +---+ +---+    Protocols    |    Equipment     |      Feed       |     DOT or       |    Feed    |    Traveling   |
|   | |   | |   |  -------------> |   Manufacturer   | --------------> |  Similar Agency  | ---------> |     Public     |
+---+ +---+ +---+                 |    or Vendor     |                 |                  |            |                |
                                  |                  |                 |                  |            |                |
                                  +------------------+                 +------------------+            +----------------+
```

**Comment by j-d-b** [comment]
I feel like option 2. is the better way to progress, because:
- It allows the producer and consumer of each feed to be explicit; and
- It allows the equipment manufacturer to only produce one feed which includes both the field devices and road events 

With this approach there must be shared models (objects and enumerated types) between the two feeds (e.g. RoadEvent). It could be accomplished without changing WZDx at all, just adding the new objects and enumerated types and schema for the SmartWorkZonesFeed. 

And we then have for "feeds" (definitions of a root-level GeoJSON document with a specific use case):

### Feeds
Feed Name | Producer | Consumer | Uses | Content
--- | --- | --- | --- | ---
WZDxFeed (existing) | Agencies responsible for managing roadways and road work, typically state and local DOTs | Traveling public via mapping companies and CAVs | Route planning; increased awareness; "put work zones on the map" | Road events
SmartWorkZonesFeed (new) | SWZ equipment manufacturers | Agencies responsible for managing roadways and permitting work, typically state and local DOTs who are also interested in producing a WZDx Feed | Aids in dynamically generating a WZDx feed; simplifies design process for agencies wanting to interface with equipment manufacturers; reduces effort for manufacturers to conform to different agencies requirements | Field devices and road events

**Comment by sergebeaudry** [comment]
Nice discussion. As a field data producer offering SWZ equipment and solutions we 2 type of data:

- Data that isnot intended to the public directly. ex: sensor raw data, full camera feed, health status, etc...
- Data that has a value to define the status of a work zone and could be useful for the public and fall within the road event portion. A few example
1.  Sensor that automatically define the boundaries of a workzone. Start, end. with the time of deployment or removal which become a key element to define if the WZ is active or not
2. Variable speed limit that could inform live on a change of speed which is an important road impact event
3. Smart ArrowBoard that will inform in real time of the location of a lane closure and if the traffic merge left- right or both side.
4. Temporary flagging or traffic signal equipment that could inform on an alternating traffic section on the road

However them alone are not the ultimate weapon as they can tell (for now) lane-level information and some administrative elements like "expected end of work, contract ID, etc.." those require human data entry of some sort.

Here a workflow I created from one of the old WZDx concept slide I've seen few years back.
![image](https://user-images.githubusercontent.com/54321625/129194750-4b6ac69d-c0b4-4305-8548-3d9dbb8d381d.png)

This looks in line with option 2 above.   That option 2 would then clarify that field data producer can provide automatically lot of information for the creation of a real-time WZDx feed.   Overtime our road events data will become richer so we need to keep the capability of becoming a "source" for almost all the road-event content and maybe more. EX: a field application could be used to enter expected duration, lane level information and all this send with real-time data from the various sensors in the field.

**Comment by j-d-b** [comment]
@sergebeaudry thank you for commenting. Would the "SmartWorkZonesFeed" contain both the data from the "planned WZ Database and the "Deploy" boxes?

**Comment by sknick-iastate** [comment]
+1 for the idea of creating a smart work zone feed to communicating data between the manufacturers and DOTs.  I lean more towards option 2 discussed above.

**Comment by j-d-b** [comment]
Please see [this comment](https://github.com/usdot-jpo-ode/wzdx/pull/204#issuecomment-921862897) in #204.

**Comment by j-d-b** [comment]
# WZDx: How/Where to Implement Field Devices

This comment further details options for how the new Field Devices should be added to the WZDx specification. I describe three options (1, 2A, 2B) for discussion with pros and cons for each.

Option 1 involves adding the field devices to the existing WZDx feed and Option 2A and 2B involve creating a new feed for the field device information, separate from the original WZDxFeed.

The decision is critical not only for adding field devices but for what WZDx represents and how new content should be added to the specification. Thus I went through the exercise of laying out all options in one comment to faciliate making a final decision.

## Option 1: all one feed, a single schema, the "WZDx Feed"

The WZDx specification describes the content and structure of a single GeoJSON file called the WZDx Feed, as it currently does in v3.1 and has since v2.0. A WZDx feed can contain both road events and field devices. Rather than a list of just `RoadEventFeature`s as in v3.1, the list (`features` array in the root `WZDxFeed` object) can contain `FieldDeviceFeature`s as well. From the allowed feature types, producers can choose what they want to include in their feed. 

#### Feeds
Feed Name | Description | Uses
--- | --- | ---
`WZDxFeed` | Describes information about events ocurring on roadways ("road events") such as work zones that change the characteristics of the roadway from the default state. WZDx is focused on describing work zones and thus also provides the capability to describe field devices deployed on roadways. | <ul><li>Allows agencies responsible for managing roadways and road work, typically state and local DOTs to provide work zone data for public use.</li><li>Details a format for providing the traveling public about road work and similar events via mapping companies and CAVs.</li><li>Allows field device manufacturers or equipment vendors to provide deployed field device information to agencies responsible for managing roadways and permitting work.</li><li>Simplifies design process for agencies wanting to interface with equipment manufacturers.</li><li>Reduces effort for manufacturers to conform to different agencies requirements.</li><li>Provides a standard format for representing work zones and field devices on a roadway.</li></ul>

### Pros
- Simple: only one feed, only one schema, flexible use
- Producers can pick and choose what to include: only road events, only field devices, or a mix
- One feed covers multiple use cases
- Least change from v3.1, v3.0, v2.0

### Cons
- As a consumer, the type of an object included in the `features` array of the feed is not explicit (could be FieldDeviceFeature or RoadEventFeature)
- WZDx doesn't define an explicit use case (in v3.1 and before: "wzdx is a format for providing work zone info for the public", now: "wzdx is a format for dissminating work zone information (to the public, or to DOTs, or however...)")
- The description of a WZDx feed is more broad and less clear what it contains

## Section 2: multiple feeds, a schema for each feed

The WZDx specification describes multiple "feeds", that is, the file structure and content for many root-level objects. Each feed is a GeoJSON [FeatureCollection](https://datatracker.ietf.org/doc/html/rfc7946#section-3.3) and there are shared objects and enumerated types between feeds. An example shared object is the feed metadata (see [RoadEventFeedInfo](/spec-content/objects/RoadEventFeedInfo)) such as contact information, but any object could be used in the schema for any feed (each feed has its own schema). The feed metadata occurs on all feeds, but what is allowed for GeoJSON features within each feed differs.

The following pros and cons apply to both **Option 2A** and **Option 2B**:

### Pros
- Allows the content of each feed to be minimal
- It is clear for the consumer what to expect in each feed
- Allows the WZDx Specification to clarify when each feed should be used (e.g. use the `WZDxFeed` to provide work zone data to the public/mapping companies)
- WZDx is more clearly modular

### Cons
- Increase in complexity
- Multiple schemas to maintain
- Producers have to provide multiple feeds (several endpoints) if they want to include all the information that WZDx covers (not necessarily a con)
- Requires updates to documentation and repository description explaining that there are multiple feeds and what each feed is for

### Option 2A: separate feeds by content

Each feed is named detailing what the root GeoJSON FeatureCollection object contains for features (e.g. RoadEventFeatures or FieldDeviceFeatures)

WZDx defines a `RoadEventFeed` (previously the `WZDxFeed`) and a `FieldDeviceFeed`. If the team/working groups define a new type of feature to represent (such as Structures, see #181), it can be added in a new feed. All feeds are types of WZDx feeds (in object oriented design, a "WZDx feed" is an interface, not an instantiable object).

In this option, WZDx defines the structure and content of several separate GeoJSON FeatureCollections each representing a collection of a specific feature type (such as `RoadEventFeature` and `FieldDeviceFeature`).

The downside of this approach is that use case of each collection of features not clearly defined. Maybe that is fine—it is certainly more flexible and open to different uses—or maybe its obfuscating the purpose of WZDx and makes it hard to know how to use each feed.

#### Feeds
Feed Name | Description | Uses
--- | --- | ---
`RoadEventFeed` (formerly named `WZDxFeed`) | Describes information about events ocurring on roadways ("road events") that have an impact to the characteristics of the roadway or temporary traffic control that leads to a change to the roadway from the default state. | <ul><li>Allows agencies responsible for managing roadways and road work, typically state and local DOTs to provide work zone data for public use.</li><li>Inform the traveling public about road work and similar events via mapping companies and CAVs.</li><li>Helps work zones on the map; increased awareness.</li></ul>
`FieldDeviceFeed` (new) | Describes high-level information about deployed field devices, such as location, status, and state. | <ul><li>Allows field device manufacturers or equipment vendors to provide deployed field device information to agencies responsible for managing roadways and permitting work, typically state and local DOTs.</li><li>Aids in dynamically generating the start/end of a work zone.</li><li>Simplifies design process for agencies wanting to interface with equipment manufacturers.</lit><li>Reduces effort for manufacturers to conform to different agencies requirements.</li></ul>

#### Pros
- The content of each feed is clear, able to be determined from the feed name
- Easily translates to a multi-endpoint REST API where each endpoint represents an entity (e.g FieldDevice, RoadEvent)
- Flexible use case, `FieldDeviceFeed` is not necessarily restricted to work zone field devices.

#### Cons
- If a producer wants to provide both field device and road event info, they have to produce two feeds
- The use case of each feed is not defined

### Option 2B: separate feeds by producer/consumer or use case

The WZDx specification defines multiple feeds which are separated by how each is intended to be used.

It could be accomplished without changing the WZDxFeed at all, just adding the new objects and enumerated types and schema for the SWZFeed. 

With this approach both the WZDxFeed (producer: DOTs/similar; consumer: the public; content: road events) and the SWZFeed (producer: vendor/manufacturer; consumer: DOTs/similar; content: work zone information (field devices and work zone road events)) contain overlapping information (e.g. they both contain road events). 

#### Feeds
Feed Name | Description | Producer | Consumer | Uses | Content
--- | --- | --- | --- | --- | ---
`WZDxFeed` (existing) | Provides information about events ocurring on roadways ("road events") such as work zones that impact the characteristics of the roadway and involve a temporary change the roadway from the default state (e.g. a lane closure). | Agencies responsible for managing roadways and road work, typically state and local DOTs | Traveling public via mapping companies and CAVs | Route planning; increased awareness; "put work zones on the map" | Road events
`SWZFeed` (new) | Provides information about field devices deployed on the roadway in work zones and optionally the definition and details of the work zone the field devices are associated with. | SWZ equipment manufacturers | Agencies responsible for managing roadways and permitting work, typically state and local DOTs who are also interested in producing a WZDx Feed | Aids in dynamically generating a WZDx feed; simplifies design process for agencies wanting to interface with equipment manufacturers; reduces effort for manufacturers to conform to different agencies requirements | Field devices and road events

#### Pros
- It allows the producer and consumer of each feed to be explicit
- Helps new implementers understand WZDx and know which feed works for their use case
- In the case that the equipment equipment manufacturer also defines road event information, they can only produce one feed which includes both the field devices and road events (though this is also true in **Option 1**)
- Prevents

#### Cons
- Possibly significant overlap in content between each feed, could just go with **Option 1** for less complexity in the specification, as it is practically the same as the `SWZFeed` proposed above (though the use case is just not clarified by the specification in option 1)
- Two feeds could potentially have the same content but different use cases (maybe confusing)
- As a consumer, the type of an object included in the `features` array of the feed is not explicit
- For producers of only field device information with no concept of road events or the work zone definition, it may not be clear that they can use the `SWZFeed` (though documentation can help clarify that)

## My Thoughts and Preference

I think Option 2A and 2B lead to a more clear specification than Option 1 because in Option 1 the feed is harder to describe and the use cases too mixed that I think it is confusing especially for newcomers to the spec.

Between 2A and 2B, especially based on the desire from some members and the specification extension subgroup to allow WZDx to represent persistent, non-temporal linear events, I think 2B makes more sense. That is because the `WZDxFeed` could be restricted to RoadEventFeatures of a certain type, such as only WorkZoneRoadEvents or DetourRoadEvents and another new feed could allow only RestrictionRoadEvents (persistent). 2B also makes it easier for new implementers to see what each feed is used for and they only have to care about writing parsing logic for one feed and about the objects that can exist within that one feed.

For what its worth, 2B is what the consensus was when the co-chairs of all subgroups met to discuss the issue in August 2021. I went through this pros/cons and clarification exercise to ensure I was ok with that decision and want to present my thoughts to ensure everyone understands what it entails so we can be confident in the decision.

**Comment by j-d-b** [comment]
> 2B keeps WZDxFeed as is—no change to content or use case or goals for this feed
>
> I may have misunderstood your description of 2A then, since I interpreted it as also remaining as is. You renamed it to `RoadEventFeed`, but the description remained the same. Why the name change? I think it's important to keep the WZDx feed as is, since it's mostly feature complete. From what i understand reading your text, the main difference between 2A/2B is:
>
> 1. Who can do what
> 2. Whether or not events/devices are completely separate, or if we want a type of feed that can encompass both type of information in one file.
 >
> But in both cases, the "normal" WZDx feed remains as is, and an additional one is added. Am I missing an important point in your suggestion here?

The difference between 2A and 2B is how to separate feeds and what the purpose of each feed is. 2A advocates for separating feeds by content, specifically the type of feature each feed contains. With 2A the purpose of a feed is to represent one type of feature. Right now that would be just field device and road events (two feeds), but as/if new types of features are added to WZDx, they would each be in a new feed. 2B separates feed by use case. The content of each feed can overlap, but the feed is defined with a specific use in mind. Neither options requires that a feed is used a certain way, but each feed is optimized for a certain use.

The WZDxFeed contains only road events (specifically work zone and detour road events) right now but is not restricted to that by design. The RoadEventFeed naming would be just a renaming for now, but would also restrict the feed to only containing road events. If new types of road events are added, even if they have nothing to do with work zones, they would have to be part of the RoadEventFeed as they are also road events. If not, it would be breaking the pattern that feeds should be separated by content.

If it helps for clarity, it would be consistent with option 2B to have a `WZDxFeed` which remains as is and a `FieldDevicesFeed` (or `SwzDevicesFeed`) which contains only the field devices. There doesn't have to be overlap in content across feeds, but by design there can be if the feeds are not defined by their content. It's not just about what goes in each feed for now, but about how feeds are defined which affects how new constructs are added to WZDx going forward.

**Comment by bzacimovic** [comment]
I have in my notes that the overall effort was moving towards Option 2B: separate feeds by producer/consumer or use cases. Yes, there could be some overlaps but trying to make one feed for everyone and everything is going to be pretty challenging.

**Comment by sknick-iastate** [comment]
I prefer option 2 and specifically option 2B.  I think it is preferable to distinguish the feeds based on the use case rather than content

**Comment by j-d-b** [comment]
Thanks, I'll move ahead with 2B for #208.

**Comment by j-d-b** [comment]
Resolved in #208.
